parser/parser.y

```diff
@@ -4144,12 +4144,6 @@ CastType:
 		x := types.NewFieldType(mysql.TypeNewDecimal)
 		x.Flen = fopt.Flen
 		x.Decimal = fopt.Decimal
-		if fopt.Flen == types.UnspecifiedLength {
```

Comment:
Have you checked other types?
If we set the default length and decimal of them in `parser.y`,
can the code be removed?